### ISSUE TITLE
No PIN requested

### ISSUE THREAD

**Issue by Pratt-Eric** [MicrosoftDocs/mixed-reality#1842]
When I attempt to deploy using Visual Studio 2019 to the hololens it never asks for a PIN. you need to provide information on how to get it to ask for a pin

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1c174da8-1ca1-0f79-9df0-608f12b041a4
* Version Independent ID: b58f67c4-dba3-b892-b97b-3ed667a19f6a
* Content: [Using Visual Studio to deploy and debug - Mixed Reality](https://docs.microsoft.com/en-us/windows/mixed-reality/using-visual-studio#feedback)
* Content Source: [mixed-reality-docs/using-visual-studio.md](https://github.com/MicrosoftDocs/mixed-reality/blob/master/mixed-reality-docs/using-visual-studio.md)
* Product: **mixed-reality**
* Technology: **mr-dev**
* GitHub Login: @pbarnettms
* Microsoft Alias: **pbarnett**

**Comment by pbarnettms** [comment]
@Pratt-Eric If you've previously associated your HoloLens device with the PC, you will not get prompted again for a PIN.  If you've never used the device with the PC, the PIN prompt should display automatically.  Are you saying you've never used the device before with the PC?  If so, what happens when you attempt to deploy from Visual Studio?  Is it deploying successfully, or are you receiving an error?  If you're receiving an error, please include the full text of the error in your reply along with the version of Visual Studio 2019 you're using (from Help -> About, which should display something like "16.2.3".)

**Comment by Pratt-Eric** [comment]
I had never associated my HoloLens with my PC before and it never asked for
a PIN. When I attempt to deploy from visual studio, I select "Release",
"x86", and "Device". When I built the project I used those settings. It all
builds just fine, but when it comes to actually deploying it says that it
wasn't able to connect to the hololens. I'm using 16.3.7.

It says the following:

"Failed to connect to device '127.0.0.1' using Universal Authentication.
Please verify the correct remote authentication mode is specified in the
project debug settings. COMException - No connection could be made because
the target machine actively refused it. (Exception from HRESULT:
0X8007274D) [0X8007274D]"

The Hololens has developer mode enabled.
My computer has developer mode enabled.

I'm NOT deploying to a remote device, but to a usb device so I'm confused
why it's trying to connect remotely.

I've already looked at
https://stackoverflow.com/questions/38578467/visual-studio-failed-deploy-to-hololens-error-dep6957-failed-to-connect-to-de

And nothing there solved my problem because I'm not deploying remotely.

The following are my build settings:
[image: image.png]

XR Settings:
[image: image.png]
Using Unity 2019.2.9f

Still trying to connect remotely regardless of me telling it to connect to
a device.

**Comment by pbarnettms** [comment]
Connecting to a device over USB is essentially a network connection to an IP address, so it is effectively a remote connection to Visual Studio, just one that happens to be to 127.0.0.1 and happens to be handled specially by a component that Visual Studio is indirectly talking to (IPoverUSB) to redirect the communication over the USB connection.

If the IPoverUSB component is not installed or not functioning, Visual Studio might be connecting to your local PC, which won't be listening for these incoming connection and could result in the error you're seeing.

Let's check a few more things:

1. If you open a web browser and navigate to http://127.0.0.1:10080/, does Device Portal load?
2. What is the output of typing 'ver' in a command prompt?
3. What is the output of typing 'sc query ipoverusbsvc' in a command prompt?  I'm particularly interested in the 'State' line, but feel free to share the full output.
4. Does the problem persist if you restart the device (e.g., using "Power" -> "Restart" in Device Portal, assuming it came up in step 1?

**Comment by Pratt-Eric** [comment]
Thank you for helping me with this.

Device Portal does not load when I navigate to that IP and port.
ver shows *Version 10.0.14393*
sc query ipoverusbsvc shows "[SC] EnumQueryServiceStatus:OpenService FAILED
1060: The specified service does not exist as an installed service."
Yes the problem persists if I restart the device. Did a hard reset on the
device as well and even cleared all paired devices.

Thank you

**Comment by Pratt-Eric** [comment]
On a separate note,
I'm doing this in a work environment and we do not have a wi-fi connection
that the hololens can connect to. My colleague can connect to the hololens
just fine and run builds on it, but I can't. I'm running the same
configuration as him.

**Comment by pbarnettms** [comment]
The issue is that IpOverUsb does not appear to be installed on your machine.  This component is installed by the Windows SDK during Visual Studio setup.  Please open the "Visual Studio Installer" from your Start menu and modify your installation to add the "USB Device Connectivity" component.  (You'll find this by clicking "Modify" for your current installation, then click to "Individual components" and scroll down to almost the bottom of the list and check "USB Device Connectivity".)  After you do this, "sc query ipoverusbsvc" should report that the service's state is "running".  Once ipoverusbsvc is reporting as running, you should be able to connect to your HoloLens.

**Comment by Pratt-Eric** [comment]
I was able to get it to connect. Seeing that I didn't have the IpToUsbSvc
on my machine I figured that would be the culprit. I re-installed windows
10 SDK manually and chose that service as an option. When installing Unity
2019 I installed Visual Studio 2019 along with it, but it didn't allow me
to select various options to install with visual studio so it just
installed the sdk without allowing me to choose what options I wanted for
it.

So when I re-installed the SDK I was able to manually pick that service,
finish up, and was able to deploy.

Thanks for your help
